src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,241 @@
+         *                    include task descriptions
+         *                    or additional information required to perform the command.
+         */
+        public ParsedCommand(CommandType commandType, String input) {
+            this.commandType = commandType;
+            this.input = input;
+            this.taskNumber = -1; // Indicates that no task number is associated with this command.
+        }
+
+        /**
+         * Constructor for commands that are associated with a specific task number.
+         * This constructor is used for commands that operate on an existing task,
+         * identified by its number.
+         *
+         * @param commandType The type of command, such as MARK, UNMARK, or DELETE.
+         * @param taskNumber  The number of the task to which the command applies.
+         */
+        public ParsedCommand(CommandType commandType, int taskNumber) {
+            this.commandType = commandType;
+            this.taskNumber = taskNumber;
+            this.input = null; // Indicates that no detailed input string is associated with this command.
+        }
+
+        /**
+         * Gets the type of the command.
+         *
+         * @return The CommandType representing the type of command.
+         */
+        public CommandType getCommandType() {
+            return commandType;
+        }
+
+        /**
+         * Gets the input string containing task details.
+         * This method returns the original input string used to create or modify a
+         * task.
+         * It is only applicable for commands that include such details.
+         *
+         * @return The input string containing task details, or null if not applicable.
+         */
+        public String getInput() {
+            return input;
+        }
+
+        /**
+         * Gets the task number associated with the command.
+         * This method returns the number of the task to which the command applies.
+         * It is only applicable for commands that operate on an existing task.
+         *
+         * @return The task number, or -1 if no task number is associated with this
+         *         command.
+         */
+        public int getTaskNumber() {
+            return taskNumber;
+        }
+    }
+
+    /**
+     * Creates a {@link Task} object based on the given command type and input.
+     * This method supports creation of Todo, Deadline, and Event tasks.
+     *
+     * @param command The command type indicating which kind of task to create.
+     * @param input   The user input string containing task details.
+     * @return A {@link Task} object or {@code null} if task creation fails.
+     */
+    public static Task createTask(CommandType command, String input) {
+        String[] parts = input.split(" ", 2);
+        Task newTask = null;
+        switch (command) {
+        case TODO:
+            if (parts[1].replaceAll("\\s", "").equals("")) {
```

Comment:
Maybe you can use a variable before the switch statement to represent parts[1].replaceAll("\\s", "") in the if statements, it can help with readability!